Source/carthage/Archive.swift

```diff
@@ -14,34 +14,67 @@ import ReactiveCocoa
 
 public struct ArchiveCommand: CommandType {
 	public struct Options: OptionsType {
-		public let frameworkName: String
 		public let outputPath: String
+		public let directoryPath: String
 		public let colorOptions: ColorOptions
+		public let frameworkNames: [String]
 
-		static func create(outputPath: String) -> ColorOptions -> String -> Options {
-			return { colorOptions in { frameworkName in
-				return self.init(frameworkName: frameworkName, outputPath: outputPath, colorOptions: colorOptions)
-			} }
+		static func create(outputPath: String) -> String -> ColorOptions -> [String] -> Options {
+			return { directoryPath in { colorOptions in { frameworkNames in
+				return self.init(outputPath: outputPath, directoryPath: directoryPath, colorOptions: colorOptions, frameworkNames: frameworkNames)
+			} } }
 		}
 
 		public static func evaluate(m: CommandMode) -> Result<Options, CommandantError<CarthageError>> {
 			return create
-				<*> m <| Option(key: "output", defaultValue: "", usage: "the path at which to create the zip file (or blank to infer it from the framework name)")
+				<*> m <| Option(key: "output", defaultValue: "", usage: "the path at which to create the zip file (or blank to infer it from the first one of the framework names)")
+				<*> m <| Option(key: "project-directory", defaultValue: NSFileManager.defaultManager().currentDirectoryPath, usage: "the directory containing the Carthage project")
 				<*> ColorOptions.evaluate(m)
-				<*> m <| Argument(usage: "the name of the built framework to archive (without any extension)")
+				<*> m <| Argument(defaultValue: [], usage: "the names of the built frameworks to archive without any extension (or blank to pick up the frameworks in the current project built by `--no-skip-current`)")
 		}
 	}
 	
 	public let verb = "archive"
-	public let function = "Archives a built framework into a zip that Carthage can use"
+	public let function = "Archives built frameworks into a zip that Carthage can use"
 
 	public func run(options: Options) -> Result<(), CarthageError> {
 		let formatting = options.colorOptions.formatting
 
+		let frameworks: [String]
+		if !options.frameworkNames.isEmpty {
+			frameworks = options.frameworkNames.map {
+				return ($0 as NSString).stringByAppendingPathExtension("framework")!
+			}
+		} else {
+			let directoryURL = NSURL.fileURLWithPath(options.directoryPath, isDirectory: true)
+			frameworks = locateProjectsInDirectory(directoryURL)
+				.flatMap(.Concat) { project in
+					return schemesInProject(project)
+						.flatMapError { error in
+							if case .NoSharedSchemes = error {
+								return .empty
```

Comment:
We should probably return the no shared schemes error here. Otherwise you get this:

```
$ carthage-dev archive
Could not find any copies of []. Make sure you're in the project’s root and that the frameworks has already been built using 'carthage build --no-skip-current'.
```